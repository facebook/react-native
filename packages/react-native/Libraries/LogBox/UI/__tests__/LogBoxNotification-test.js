/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../../jest/renderer');
const LogBoxLog = require('../../Data/LogBoxLog').default;
const LogBoxNotification = require('../LogBoxNotification').default;
const React = require('react');

// Mock child components because we are interested in snapshotting the behavior
// of `LogBoxNotification`, not its children.
jest.mock('../LogBoxButton', () => ({
  __esModule: true,
  default: 'LogBoxButton',
}));
jest.mock('../LogBoxNotificationCountBadge', () => ({
  __esModule: true,
  default: 'LogBoxNotificationCountBadge',
}));
jest.mock('../LogBoxNotificationDismissButton', () => ({
  __esModule: true,
  default: 'LogBoxNotificationDismissButton',
}));
jest.mock('../LogBoxNotificationMessage', () => ({
  __esModule: true,
  default: 'LogBoxNotificationMessage',
}));

const log = new LogBoxLog({
  level: 'warn',
  isComponentError: false,
  message: {
    content: 'Some kind of message',
    substitutions: [],
  },
  stack: [],
  category: 'Some kind of message',
  componentStack: [],
});

describe('LogBoxNotification', () => {
  it('should render log', async () => {
    const output = await render.create(
      <LogBoxNotification
        log={log}
        totalLogCount={1}
        level="warn"
        onPressOpen={() => {}}
        onPressDismiss={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
