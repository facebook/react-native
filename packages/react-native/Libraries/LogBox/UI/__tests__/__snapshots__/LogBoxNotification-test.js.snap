// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxNotification should render log 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "height": 48,
      "justifyContent": "center",
      "marginTop": 0.5,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "rgba(51, 51, 51, 1)",
        "pressed": "rgba(51, 51, 51, 0.9)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "height": 48,
        "justifyContent": "center",
        "marginTop": 0.5,
        "paddingHorizontal": 12,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderRadius": 8,
          "flexBasis": "auto",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 0,
        }
      }
    >
      <LogBoxNotificationCountBadge
        count={1}
        level="warn"
      />
      <LogBoxNotificationMessage
        message={
          Object {
            "content": "Some kind of message",
            "substitutions": Array [],
          }
        }
      />
      <LogBoxNotificationDismissButton
        onPress={[Function]}
      />
    </View>
  </LogBoxButton>
</View>
`;
